firefox_ui_tests/remote/security/test_safe_browsing_warning_pages.py

```diff
@@ -0,0 +1,101 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import time
+
+from marionette_driver import (
+    By,
+    expected
+)
+from marionette_driver.errors import NoSuchElementException
+
+from firefox_ui_harness.testcase import FirefoxTestCase
+
+
+class TestSafeBrowsingWarningPages(FirefoxTestCase):
+
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+
+        self.urls = [
+            # Phishing URL
+            'https://www.itisatrap.org/firefox/its-a-trap.html',
+            # Malware URL
+            'https://www.itisatrap.org/firefox/its-an-attack.html'
+            ]
+
+        self.prefs.set_pref('browser.safebrowsing.enabled', True)
+        self.prefs.set_pref('browser.safebrowsing.malware.enabled', True)
+
+        # Give the browser a little time, because SafeBrowsing.jsm takes a
+        # while between start up and adding the example urls to the db.
+        # hg.mozilla.org/mozilla-central/file/46aebcd9481e/browser/base/content/browser.js#l1194
+        time.sleep(3)
+
+    def test_warning_pages(self):
+        with self.marionette.using_context("content"):
+            for unsafe_page in self.urls:
+                # Load a test page and test the get me out button
+                self.marionette.navigate(unsafe_page)
+                self.check_get_me_out_of_here_button(unsafe_page)
+
+                # Load the test page again and test the report button
+                self.marionette.navigate(unsafe_page)
+                self.check_report_button(unsafe_page)
+
+                # Load the test page again and test the ignore warning button
+                self.marionette.navigate(unsafe_page)
+                self.check_ignore_warning_button(unsafe_page)
+
+    def check_get_me_out_of_here_button(self, unsafe_page):
+        button = self.marionette.find_element(By.ID, "getMeOutButton")
+
+        # Wait for the DOM to receive events
+        time.sleep(1)
+        button.click()
+
+        self.wait_for_condition(lambda mn: self.browser.default_homepage in mn.get_url())
+
+    def check_report_button(self, unsafe_page):
+        button = self.marionette.find_element(By.ID, "reportButton")
+
+        # Wait for the DOM to receive events
+        time.sleep(1)
+        button.click()
+
+        # Wait for page load so we can record and verify the url
+        self.wait_for_condition(lambda mn: mn.execute_script(
+            "return document.readyState == 'complete'"))
```

Comment:
This causes a race-condition if the page load is a bit delayed. So it would return immediately. Wait for an element of the target page for now instead.